### ISSUE TITLE
< > not escaped in anchor reftext definition

### ISSUE THREAD

**Issue by aranega** [asciidoctor/asciidoctor#1694]
Hi there,

I have a small issue using anchor captions. When `<` or `>` are used in caption for anchor definition, the symbol are not translated in `&lt;` or `&gt;` resulting in rendering errors in HTML and parser warnings with the PDF backend.

E.g., this code:

```
See <<myuuid>>

[[myuuid, My ref title with <T>]]
== This is my real title
```

is translated as this:

```
<p>We can see the issue in <a href="#myuuid">My ref title with <T></a></p>
```

I use Asciidoctor 1.5.4

**Comment by mojavelinux** [comment]
This does seem problematic. I'll classify this as a bug.

**Comment by mojavelinux** [comment]
This actually raises two questions.

1. Should the reftext be stored with substitutions applied, or should the substitutions be applied in the converter?
2. What substitutions should be applied to the reftext

If we think about the reftext (i.e., xreflabel) as analogous to the title, then I think we have our answer to both questions. We should give the reftext the same treatment as the block title. That means applying title substitutions to the value when it is assigned to the block node. That's how I'll proceed.

**Comment by mojavelinux** [comment]
The other reason we kind of have to go in this direction is because we don't know downstream whether the label came from the title or the reftext. So we really need to give them the same treatment upstream.

**Comment by mojavelinux** [comment]
This issue will be resolved by #2220 as part of the fix for #858.

To answer the questions from above, attribute references are substituted before the reftext is stored in the document catalog and reftext subs (a new substitution group) are applied when the reftext is inserted into the output at the location of a cross reference.

**Comment by mojavelinux** [comment]
Resolved by 8e80317dee56483e48a07451bff0cd14f0bfd59c.

Test case: https://github.com/asciidoctor/asciidoctor/blob/8e80317dee56483e48a07451bff0cd14f0bfd59c/features/xref.feature#L7-L37
